src/deriver/consumer.py

```
@@ -3,57 +3,37 @@
     # Validate payload structure and types before processing
     try:
-        validated_payload = DeriverQueuePayload(**payload)
+        task_type = payload.get("task_type")
+        if task_type == "representation":
+            validated_payload = RepresentationPayload(**payload)
+        elif task_type == "summary":
+            validated_payload = SummaryPayload(**payload)
+        else:
+            raise ValueError(f"Invalid task_type: {task_type}")
     except ValidationError as e:
         logger.error("Invalid payload received: %s. Payload: %s", str(e), payload)
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)